recipe/meta.yaml

```
@@ -1,124 +1,166 @@
 {% set version = "0.16.0" %}
 {% set filename = "apache-arrow-" + version + ".tar.gz" %}
 {% set checksum = "261992de4029a1593195ff4000501503bd403146471b3168bd2cc414ad0fb7f5" %}
+{% set number = "4" %}
+{% set cuda_enabled = cuda_compiler_version is not undefined and cuda_compiler_version == '9.2' %}
+{% set build_ext_version = "1.0.0" %}
+{% set build_ext = "cuda" if cuda_enabled else "cpu" %}
 
 package:
-  name: arrow-cpp
-  version: {{ version }}
+  name: arrow-cpp-ext
 
 source:
   fn: {{ filename }}
   url: https://dist.apache.org/repos/dist/release/arrow/arrow-{{ version }}/{{ filename }}
   sha256: {{ checksum }}
 
 build:
-  number: 3
-  skip: true  # [win and vc<14]
+  skip: true  # [(win and vc<14) or (linux and cuda_compiler_version is not undefined and cuda_compiler_version != '9.2' and cuda_compiler_version != 'None')]
+  number: {{ number }}
   run_exports:
-    - {{ pin_subpackage("arrow-cpp", max_pin="x.x.x") }}
+    - {{ pin_subpackage("arrow-cpp", max_pin="x.x.x") }}  # is this correct and the correct place for it?
 
-requirements:
-  build:
-    - cmake
-    - autoconf  # [unix]
-    - ninja
-    - make  # [unix]
-    - {{ compiler('c') }}
-    - {{ compiler('cxx') }}
-  host:
-    - aws-sdk-cpp
-    - boost-cpp
-    - brotli
-    - bzip2
-    - c-ares
-    - gflags
-    - glog
-    - grpc-cpp
-    - libprotobuf
-    - clangdev 7.*  # [not (aarch64 or ppc64le)]
-    - llvmdev 7.*   # [not (aarch64 or ppc64le)]
-    - lz4-c
-    - numpy 1.16.*
-    - orc  # [unix]
-    - python
-    - rapidjson
-    - re2
-    - snappy
-    - thrift-cpp >=0.11
-    - zlib
-    - zstd
+outputs:
+  - name: arrow-cpp-proc
+    version: {{ build_ext_version }}
+    build:
+      number: 0
+      string: "{{ build_ext }}"
+    test:
+      commands:
+        - exit 0
+    about:
+      home: http://github.com/apache/arrow
+      license: Apache-2.0
+      license_file:
+        - LICENSE.txt
+      summary: 'A meta-package to select Arrow build variant'
 
-  run:
-    - {{ pin_compatible('numpy', lower_bound='1.16') }}
-    - aws-sdk-cpp
-    - boost-cpp
-    - brotli
-    - c-ares
-    - gflags
-    - glog
-    - grpc-cpp
-    - lz4-c
-    - python
-    - re2
-    - snappy
-    - zlib
-    - zstd
+  - name: arrow-cpp
+    script: build-arrow.sh  # [not win]
+    script: bld.bat         # [win]
+    version: {{ version }}
+    build:
+      number: {{ number }}
+      string: "py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ number }}_{{ build_ext }}"
+      skip: true  # [win and vc<14]
+      skip: true  # [linux and cuda_compiler_version is not undefined and cuda_compiler_version != '9.2' and cuda_compiler_version != 'None']
```

We might be able to simplify this like so. IOW other platforms where CUDA is not yet supported (`undefined`), CPU-only builds on platforms that do support CUDA (`None`), and CUDA 9.2 are permitted. Otherwise we skip all other CUDA versions. Please let me know if that matches the intent or if I'm missing something here.

```suggestion
      skip: true  # [cuda_compiler_version not in (undefined, "None", "9.2")]
```

---

The suggested skip condition is as intended, so let's use it.